apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -1315,113 +1311,25 @@ func (m *ModelManagerAPI) ModifyModelAccess(ctx context.Context, args params.Mod
 			result.Results[i].Error = apiservererrors.ServerError(errors.Annotate(err, "could not modify model access"))
 			continue
 		}
+		err = m.accessService.UpdatePermission(ctx, access.UpdatePermissionArgs{
+			AccessSpec: permission.AccessSpec{
+				Target: permission.ID{
+					ObjectType: permission.Model,
+					Key:        modelTag.Id(),
+				},
+				Access: modelAccess,
+			},
+			AddUser: true,
+			ApiUser: m.apiUser.Id(),
+			Change:  permission.AccessChange(arg.Action),
+			Subject: targetUserTag.Id(),
```

Comment:
Can you test calling this via the client on version 3.x and this one, with an external user (user@domain)?

If it works on 3.x for an external user, we will ultimately want to create such user entries if they do not already exist, but we can handle it later.